ui-admin/src/study/settings/study/KitSettings.tsx

```diff
@@ -0,0 +1,135 @@
+import React, {
+  useEffect,
+  useState
+} from 'react'
+import {
+  StudyEnvironmentConfig,
+  StudyEnvParams
+} from '@juniper/ui-core'
+import InfoPopup from 'components/forms/InfoPopup'
+import { useUser } from 'user/UserProvider'
+import Select from 'react-select'
+import { Button } from 'components/forms/Button'
+import LoadingSpinner from 'util/LoadingSpinner'
+import { doApiLoad } from 'api/api-utils'
+import Api from 'api/api'
+import { Store } from 'react-notifications-component'
+import { successNotification } from 'util/notifications'
+import { StudyEnvContextT } from 'study/StudyEnvironmentRouter'
+import { LoadedPortalContextT } from 'portal/PortalProvider'
+import {
+  InfoCard,
+  InfoCardHeader,
+  InfoCardTitle
+} from 'components/InfoCard'
+
+export const KitSettings = (
+  {
+    studyEnvContext,
+    portalContext,
+    config,
+    updateConfig
+  }: {
+    studyEnvContext: StudyEnvContextT,
+    portalContext: LoadedPortalContextT,
+    config: StudyEnvironmentConfig,
+    updateConfig: (key: keyof StudyEnvironmentConfig, value: unknown) => void
+  }) => {
+  const { user } = useUser()
+
+  const studyEnvParams: StudyEnvParams = {
+    portalShortcode: studyEnvContext.portal.shortcode,
+    studyShortcode: studyEnvContext.study.shortcode,
+    envName: studyEnvContext.currentEnv.environmentName
+  }
+
+  const loadAllowedKitTypes = async () => {
+    const allowedKitTypes = await Api.fetchAllowedKitTypes(studyEnvParams)
+    const allowedKitTypeOptions = allowedKitTypes.map(kt => ({ value: kt.name, label: kt.displayName }))
+    setKitTypeOptions(allowedKitTypeOptions)
+  }
+
+  const loadConfiguredKitTypes = async () => {
+    const selectedKitTypes = await Api.fetchKitTypes(studyEnvParams)
+    setSelectedKitTypes(selectedKitTypes.map(kt => ({ value: kt.name, label: kt.displayName })))
+  }
+
+  useEffect(() => {
+    loadAllowedKitTypes()
+    loadConfiguredKitTypes()
+  }, [studyEnvContext.currentEnvPath])
```

Comment:
I don't think it's worth the complexity to track separate loading states for allowed and configured kit types--opens the possibility for more edge cases.  Also, it's clearer if the state more-or-less reflects what was fetched from the server, and the rendering methods handle transforming it into options

```suggestion
  const {isLoading} = useLoadingEffect(() => {
     const [selectedKitTypes, allowedKitTypes] = await Promise.all([
       Api.fetchKitTypes(studyEnvParams),
       Api.fetchAllowedKitTypes(studyEnvParams)
       ])
       setKitTypes(selectedKitTypes)
       setAllowedKitTypes(allowedKitTypes)
  }, [studyEnvContext.currentEnvPath])
```